.github/workflows/ci.yaml

```diff
@@ -0,0 +1,74 @@
+name: Actions_Pipeline
+
+on:
+  push:
+    branches: [ features/add_implementation ]
+  pull_request:
+    branches: [ features/add_implementation ]
+
+env:
+  IMAGE_NAME: image-angular-react-starter
+  GH_TOKEN: ${{ github.token }}
+
+jobs:
+  pipeline_of_build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Get repository code
+        uses: actions/checkout@v3
+
+      - name: Installing packages Angular
+        run: cd ${{ github.workspace }}/angular && npm install
+      - name: Installing packages React
+        run: cd ${{ github.workspace }}/react && npm install
+
+      - name: Run test
+        uses: GabrielBB/xvfb-action@v1
+        with:
+          run: cd ${{ github.workspace }}/angular && npm run test
+
+      - name: Build project
+        run: cd ${{ github.workspace }}/angular && npm run build

```

Comment:
this step has to be executed only if test step passed.

Reply:
otherwise you will push the image with a broken code which will then get deployed